File: Java/util/TimeOfDay.java
package util;

public final class TimeOfDay {
    public final int hour;
    public final int minute;
    public final int second;

    public TimeOfDay(int hour, int minute, int second)
    {
        this.hour = hour;
        this.minute = minute;
        this.second = second;
    }

    public static TimeOfDay midnight()
    {
        return new TimeOfDay(0, 0, 0);
    }
}

File: C#/TollCalculator/TollCalculator.Lib/Models/TimeOfDay.cs
using System;

namespace TollCalculator.Lib.Models
{
    public class TimeOfDay
    {
        public int Hour { get; set; }
        public int Minute { get; set; }

        public TimeOfDay(int hour, int minute)
        {
            Hour = hour;
            Minute = minute;
        }

        public DateTime ToDateTime(DateTime reference)
        {
            return new DateTime(reference.Year, reference.Month, reference.Day, Hour, Minute, reference.Second,
                reference.Millisecond, reference.Kind);
        }
    }
}